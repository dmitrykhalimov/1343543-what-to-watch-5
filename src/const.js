export const ratingRanks = [
  {
    'Bad': 2.9
  },
  {
    'Normal': 4.9
  },
  {
    'Good': 7.9
  },
  {
    'Very good': 9.9
  },
  {
    'Awesome': 10
  }
];
