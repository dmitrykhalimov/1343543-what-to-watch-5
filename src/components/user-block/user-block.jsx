import React from "react";
import UserAvatar from "../user-avatar/user-avatar";
import {AuthorizationStatus} from "../../const";
import UserSignIn from "../user-sign-in/user-sign-in";
import {validUserData} from "../../utils/props";

const UserBlock = (props) => {
  const {authorizationStatus, avatarUrl} = props.userData;
  return (
    <div className="user-block">
      {authorizationStatus === AuthorizationStatus.AUTH
        ? <UserAvatar avatarUrl = {avatarUrl}/>
        : <UserSignIn/>
      }
    </div>
  );
};

UserBlock.propTypes = {
  userData: validUserData,
};

export default UserBlock;

