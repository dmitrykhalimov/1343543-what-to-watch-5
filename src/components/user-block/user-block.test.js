import React from "react";
import renderer from "react-test-renderer";
import {UserBlock} from "./user-block";
import {MemoryRouter} from 'react-router-dom';
import {TEST_MOCK_STORE} from "../../const";

describe(`UserBlock`, () => {
  it(`Should UserBlock render correctly, isPlaying`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <UserBlock
              userData={TEST_MOCK_STORE.user}
            />
          </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
