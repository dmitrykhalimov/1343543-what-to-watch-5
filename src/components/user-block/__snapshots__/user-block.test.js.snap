// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock Should UserBlock render correctly, isPlaying 1`] = `
<div
  className="user-block"
>
  <a
    href="/mylist"
    onClick={[Function]}
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="https://i.kym-cdn.com/photos/images/newsfeed/000/607/429/84b.jpg"
        width="63"
      />
    </div>
  </a>
</div>
`;
